### PR TITLE
lightning: duplicate entry should be "conflict error" in tidb backend

### LINKED ISSUE #44608: lightning tidb backend should record "duplicate entry" error as conflict error
## Enhancement
as title. previously, it's recorded as type error

### PR DESCRIPTION
**Pull request by lance6716**
### What problem does this PR solve?

Issue Number: close #44608

Problem Summary:

### Check List

Tests 

- [x] Unit test
- [x] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [x] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: br/tests/lightning_config_max_error/run.sh [hunk 1]
```diff
@@ -79,3 +79,18 @@ check_contains "COUNT(*): ${duplicated_row_count}"
 # Check remaining records in the target table
 run_sql 'SELECT COUNT(*) FROM mytest.testtbl'
 check_contains "COUNT(*): ${uniq_row_count}"
+
+# Check tidb backend record duplicate entry in conflict_error_v2 table
+run_sql 'DROP TABLE IF EXISTS lightning_task_info.conflict_error_v1'
```
Comment (D3Hunter) [inline br/tests/lightning_config_max_error/run.sh:84]: drop `conflict_error_v2`?

**Review by D3Hunter — APPROVED** [review]

**Review by GMHDBJD — APPROVED** [review]
LGTM

**Comment by lance6716** [comment]
/hold

after discuss, the whole behaviour should not be changed in v7.2

**Review by GMHDBJD — APPROVED** [review]
LGTM

**Pull request merged**